integrations/src/test/java/io/jenkins/plugins/casc/core/CloudSupportTest.java

```diff
@@ -0,0 +1,170 @@
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.slaves.AbstractCloudComputer;
+import hudson.slaves.AbstractCloudSlave;
+import hudson.slaves.EphemeralNode;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.PretendSlave;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class CloudSupportTest {
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    @ConfiguredWithCode("CloudSupportTest.yml")
+    public void should_have_nodes_configured() {
+        Assert.assertEquals("Base nodes not found", 2, j.jenkins.getNodes().size());
+    }
+
+    @Test
+    @ConfiguredWithCode("CloudSupportTest.yml")
+    public void should_remove_normal_nodes_configured_after_reload() throws Exception {
+        Assert.assertEquals("Base nodes not found", 2, j.jenkins.getNodes().size());
+
+        final Node slave = new StaticPretendSlave();
+        j.jenkins.addNode(slave);
+
+        ConfigurationAsCode.get().configure(this.getClass().getResource("CloudSupportTest.yml").toString());
+        Assert.assertEquals("Base nodes not found", 2, j.jenkins.getNodes().size());
+    }
+
+    @Test
+    @ConfiguredWithCode("CloudSupportTest.yml")
+    public void should_keep_cloud_no_instantiable_nodes_configured_after_reload() throws Exception {
+        Assert.assertEquals("Base nodes not found", 2, j.jenkins.getNodes().size());
+        Assert.assertNotNull("Slave 1", j.jenkins.getNode("agent1"));
+        Assert.assertNotNull("Slave 1", j.jenkins.getNode("agent2"));
+
+        final Node slave = new Cloud1PretendSlave();
+        j.jenkins.addNode(slave);
+
+        ConfigurationAsCode.get().configure(this.getClass().getResource("CloudSupportTest.yml").toString());
+        Assert.assertEquals("Cloud nodes not found", 3, j.jenkins.getNodes().size());
+        Assert.assertNotNull("Slave 1", j.jenkins.getNode("agent1"));
+        Assert.assertNotNull("Slave 1", j.jenkins.getNode("agent2"));
+        Assert.assertNotNull("Slave cloud", j.jenkins.getNode("testCloud"));
+    }
+
+    @Test
+    @ConfiguredWithCode("CloudSupportTest.yml")
+    public void should_keep_cloud_ephemeral_nodes_configured_after_reload() throws Exception {
+        Assert.assertEquals("Base nodes not found", 2, j.jenkins.getNodes().size());
+        Assert.assertNotNull("Slave 1", j.jenkins.getNode("agent1"));
+        Assert.assertNotNull("Slave 1", j.jenkins.getNode("agent2"));
+
+        final Node slave = new Cloud2PretendSlave();
+        j.jenkins.addNode(slave);
+
+        ConfigurationAsCode.get().configure(this.getClass().getResource("CloudSupportTest.yml").toString());
+        Assert.assertEquals("Cloud nodes not found", 3, j.jenkins.getNodes().size());
```

Comment:
normal practice is to add a static import for assert* methods, makes it more natural to read

```suggestion
       assertEquals("Cloud nodes not found", 3, j.jenkins.getNodes().size());
```

Reply:
Yup I forgot that in the end of the day..